charts/descheduler/templates/_helpers.tpl

```diff
@@ -87,8 +87,10 @@ Leader Election
 {{- if .Values.leaderElection.resourceName }}
 - --leader-elect-resource-name={{ .Values.leaderElection.resourceName }}
 {{- end }}
-{{- if .Values.leaderElection.resourceNamescape }}
-- --leader-elect-resource-namespace={{ .Values.leaderElection.resourceNamescape }}
+{{- if .Values.leaderElection.resourceNamespace }}
+- --leader-elect-resource-namespace={{ .Values.leaderElection.resourceNamespace }}
+{{- else if .Values.leaderElection.resourceNamescape }}
+{{- fail "'leaderElection.resourceNamescape' is deprecated. instead use 'leaderElection.resourceNamespace'" }}
```

Comment:
We should not fail when a param is deprecated. Would it make more sense to print a warning but still take `resourceNamescape into account?

Reply:
I don't know of a way to print warning in helm.

I think this is a trivial change to prompt users to correct. we could continue to support both indefinitely

Reply:
`resourceNamespace` was introduced in Feb 2022. No way to tell how many users rely on that parameter. I am for deprecating `resourceNamespace` in the release notes, supporting both for 2 release and removing the deprecated param then.

Reply:
supporting both now, let me know what you think